src/psyclone/psyGen.py

```diff
@@ -2721,34 +2696,28 @@ def dag_name(self):
 
     @property
     def module_inline(self):
+        '''
+        :returns: whether or not this kernel is being module-inlined.
+        :rtype: bool
+        '''
         return self._module_inline
 
     @module_inline.setter
     def module_inline(self, value):
         '''
         Setter for whether or not to module-inline this kernel.
 
-        :param bool value: Whether or not to module-inline this kernel.
-        :raises NotImplementedError: if module-inlining is enabled and the \
-                                     kernel has been transformed.
+        :param bool value: whether or not to module-inline this kernel.
         '''
         # Check all kernels in the same invoke as this one and set any
         # with the same name to the same value as this one. This is
         # required as inlining (or not) affects all calls to the same
         # kernel within an invoke. Note, this will set this kernel as
         # well so there is no need to set it locally.
-        if value and self.modified:
-            # TODO #229. Kernel in-lining is currently implemented via
-            # manipulation of the fparser1 Parse Tree while
-            # transformations work with the fparser2 Parse Tree-derived
-            # PSyIR.  Therefore there is presently no way to inline a
-            # transformed kernel.
-            raise NotImplementedError(
-                "Cannot module-inline a transformed kernel ({0}).".
-                format(self.name))
         my_schedule = self.ancestor(InvokeSchedule)
         for kernel in my_schedule.walk(Kern):
             if kernel.name == self.name:
+                # pylint: disable=protected-access
                 kernel._module_inline = value
```

Comment:
There's a setter for this so you can just do `kernel.module_inline = value`  I think?

Reply:
We are already inside this setter :) The problem is that it applies it to all kernels with the same name and if we would call the setter again it will trigger an infinite recursive call. I added a comment to make this clear.

Reply:
Ahem, I knew that ;-) I see the problem now. We could protect it with `if kernel is not self` and then separately set `self._module_inline = value`. I'm not sure which is best - I'll let you play and decide.

Reply:
Removed protected access.